lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -850,24 +852,32 @@ void LCIStrategicPlugin::planWhenWAITING(const cav_srvs::PlanManeuversRequest& r
       wm_->routeTrackPos(stop_line.get().front().basicPoint2d()).downtrack;
 
   ROS_DEBUG("traffic_light_down_track %f", traffic_light_down_track);
+
+  double entering_time = current_state.stamp.toSec();
   
-  auto current_light_state_optional = traffic_light->predictState(lanelet::time::timeFromSec(current_state.stamp.toSec()));
-  ROS_DEBUG_STREAM("WAITING STATE: requested time to check: " << std::to_string(req.header.stamp.toSec()));
-  ROS_DEBUG_STREAM("WAITING STATE: requested time to CURRENT STATE check: " << std::to_string(current_state.stamp.toSec()));
+  auto current_light_state_optional = traffic_light->predictState(lanelet::time::timeFromSec(entering_time));
+
+  ROS_DEBUG_STREAM("WAITING STATE: requested time to req.header.stamp check: " << std::to_string(req.header.stamp.toSec()));
+  ROS_DEBUG_STREAM("WAITING STATE: requested time to CURRENT STATE check: " << std::to_string(entering_time));
   
-  if (!validLightState(current_light_state_optional, current_state.stamp))
+  if (!validLightState(current_light_state_optional, ros::Time(entering_time)))
     return;
 
-  auto bool_optional_late_certainty = canArriveAtGreenWithCertainty(current_state.stamp, traffic_light, true, false);
+  auto bool_optional_late_certainty = canArriveAtGreenWithCertainty(ros::Time(entering_time), traffic_light, true, false);
```

Comment:
To me these changes in time read like we are evaluating when the vehicle crosses the stop bar from a stop. Previously the logic seemed to wait until the light actually changed. Does this cause the car to start moving while the light is still red? That would be unsafe in my opinion.

Reply:
Actually these changes ended up being cosmetic as they are doign exactly the same thing as uc2. Any changes inside the function planWhenWAITING should be reverted. 

You probably missed that entering_time was assigned current_state.stamp.toSec(); so it was just a renaming. 

And this condition is never satisfied
   if (config_.enable_carma_streets_connection && entering_time > current_state.stamp.toSec()) //uc3
    should_enter = false; 

so it is same as not being there and can be removed.

Reply:
So the current behavior is still just "enter when green is ON if you are waiting at the stop_line (doesn't matter the ET sent by carma-street, which is what we discussed with Misheel, Paul and Saeid)"